src/firefly/js/ui/TimePanel.jsx

```diff
@@ -19,72 +19,64 @@ const iconMap = {'calendar': {icon: CALENDAR, title: 'Show the calendar for sele
-
-        const outsideWidth = inputWidth + 6;
-        const timePart = iconField ? (<div style={{position: 'relative', width: outsideWidth}}>
-                                            {timeField}
-                                            {iconField}
-                                      </div>)
-                                    : (timeField);
-
-        const newFeedbackStyle = clone(feedbackStyle, {width: inputWidth});
-        const lStyle = clone(labelStyle, {whiteSpace:'nowrap'});
-        const labelDiv = (<div style={lStyle}>{label}</div>);
-        const timeDiv = (
-            <div>
-                {timePart}
-                <TimeFeedback {...{showHelp, feedback, feedbackStyle: newFeedbackStyle, examples, timeMode}}/>
-            </div>
-        );
-        const flexDirection = (labelPosition && labelPosition === 'top') ? 'column' : 'row';
-
-        return (
-            <div style={{display: 'flex', flexDirection}}>
-                {labelDiv}
-                {timeDiv}
-            </div>
-        );
-    }
+function TimePanelView({showHelp, feedback, feedbackStyle, examples, label, labelStyle, labelPosition,
+                           valid, message, onChange, value, icon, onClickIcon, tooltip = 'select time',
+                           inputStyle, wrapperStyle, inputWidth, timeMode=ISO}) {
+    const ImagePadding = 3;
+
+    const iconStyle = {
+        position: 'absolute',
+        top: 1,
+        right: 1,
+        padding: ImagePadding,
+        cursor: 'pointer'
+    };
+
+    const iconField = icon && has(iconMap, icon) ?
+        (<div style={iconStyle}>
+            <img
+              title={iconMap[icon].title}
+              src={iconMap[icon].icon}
+              onClick={() => {onClickIcon && onClickIcon();}}/>
+         </div>) : null;
+
+    const spaceForImage = 16+ImagePadding*2;
+    const newInputStyle = Object.assign({paddingRight: iconField ? spaceForImage : 2}, inputStyle,
+                                        {width: iconField ? inputWidth - spaceForImage+2 : inputWidth});
+    const placeHolder = timeMode === ISO ? 'YYYY-MM-DD HH:mm:ss' : 'float number .... ';
+    const newWrapperStyle = clone(wrapperStyle, (iconField ? {width: '%100'} : {}));
+    const inputFields = {
+        valid, visible: true, message, onChange,
+        value, tooltip, wrapperStyle: newWrapperStyle, style: newInputStyle,
+        placeHolder
+    };
+
+    const timeField =  (<InputFieldView {...inputFields} />);
+
+    const outsideWidth = inputWidth + 6;
+    const timePart = iconField ? (<div style={{position: 'relative', width: outsideWidth}}>
+                                        {timeField}
+                                        {iconField}
+                                  </div>)
+                                : (timeField);
+
+    const newFeedbackStyle = clone(feedbackStyle, {width: inputWidth});
+    const lStyle = clone(labelStyle, {whiteSpace:'nowrap'});
```

Comment:
We are getting away from using `clone` or `Object.assign`.  These two lines should be:
```javascript
 const newFeedbackStyle = {...feedbackStyle, width: inputWidth};
 const lStyle = {...labelStyle, whiteSpace:'nowrap'};
```